packages/editor-ui/src/components/Node/NodeCreator/composables/useViewStacks.ts

```diff
@@ -99,34 +122,147 @@ export const useViewStacks = defineStore('nodeCreatorViewStacks', () => {
 	const globalSearchItemsDiff = computed<INodeCreateElement[]>(() => {
-		const stack = viewStacks.value[viewStacks.value.length - 1];
+		const stack = getLastActiveStack();
 		if (!stack?.search) return [];
 
 		const allNodes = nodeCreatorStore.mergedNodes.map((item) => transformNodeType(item));
-		const globalSearchResult = extendItemsWithUUID(searchNodes(stack.search || '', allNodes));
+		// Apply filtering for AI nodes if the current view is not the AI root view
+		const filteredNodes = isAiRootView(stack) ? allNodes : filterOutAiNodes(allNodes);
+
+		let globalSearchResult: INodeCreateElement[] = extendItemsWithUUID(
+			searchNodes(stack.search || '', filteredNodes),
+		);
+		if (isAiRootView(stack)) {
+			globalSearchResult = groupIfAiNodes(globalSearchResult);
+		}
+
+		const filteredItems = globalSearchResult.filter((item) => {
+			return !activeStackItems.value.find((activeItem) => {
+				if (activeItem.type === 'section') {
+					const matchingSectionItem = activeItem.children.some(
+						(sectionItem) => sectionItem.key === item.key,
+					);
+					return matchingSectionItem;
+				}
+
+				return activeItem.key === item.key;
+			});
+		});
+
+		// Filter out empty sections if all of their children are filtered out
+		const filteredSections = filteredItems.filter((item) => {
+			if (item.type === 'section') {
+				const hasVisibleChildren = item.children.some((child) =>
+					activeStackItems.value.some((filteredItem) => filteredItem.key === child.key),
+				);
 
-		return globalSearchResult.filter((item) => {
-			return !activeStackItems.value.find((activeItem) => activeItem.key === item.key);
+				return hasVisibleChildren;
+			}
+
+			return true;
 		});
+
+		return filteredSections;
 	});
 
 	const itemsBySubcategory = computed(() => subcategorizeItems(nodeCreatorStore.mergedNodes));
 
+	function isAiRootView(stack: ViewStack) {
+		return stack.rootView === AI_NODE_CREATOR_VIEW;
+	}
+
+	function groupIfAiNodes(items: INodeCreateElement[], sortAlphabetically = true) {
+		const aiNodes = items.filter((node): node is NodeCreateElement => isAINode(node));
+
+		if (aiNodes.length > 0) {
+			const sectionsMap = new Map<string, NodeViewItemSection>();
+			aiNodes.forEach((node) => {
+				const section = node.properties.codex?.subcategories?.[AI_SUBCATEGORY]?.[0];
+
+				if (section) {
+					const currentItems = sectionsMap.get(section)?.items ?? [];
+					const isSubnodesSection =
+						!node.properties.codex?.subcategories?.[AI_SUBCATEGORY].includes(
+							AI_CATEGORY_ROOT_NODES,
+						);
+
+					sectionsMap.set(section, {
+						key: section,
+						title: isSubnodesSection
+							? `${section} (${i18n.baseText('nodeCreator.subnodes')})`
+							: section,
+						items: [...currentItems, node.key],
+					});
+				}
+			});
+
+			const nonAiNodes = difference(items, aiNodes);
+			const nonAiTriggerNodes = nonAiNodes.filter(
+				(item) => item.type === 'node' && item.properties.name.toLowerCase().includes('trigger'),
```

Comment:
Should we check `item.properties.group.includes('trigger')` instead of `name`? also found this check in other NodeCreator code (not related to this PR)

Reply:
Good point. We already have a helper in the NodeTypes store that does this, so I've replaced the usage here.